python/ray/experimental/channel/nccl_group.py

```diff
@@ -188,22 +253,40 @@ def recv(
             raise RayChannelError("NCCL group has been destroyed.")
         assert allocator is not None, "NCCL group requires a tensor allocator"
         buf = allocator(shape, dtype)
-        self._comm.recv(
-            self.nccl_util.get_tensor_ptr(buf),
-            buf.numel(),
-            self.nccl_util.get_nccl_tensor_dtype(buf),
-            peer_rank,
-            self._cuda_stream.ptr,
-        )
 
-        # Buffer values are undefined if NCCL ops are aborted. Therefore, we
-        # need to synchronize here and check that the channel is still open to
-        # ensure that the receive buffer is valid.
-        # TODO(swang): Avoid CUDA synchronization.
-        self._cuda_stream.synchronize()
-        if self._closed:
-            raise RayChannelError("NCCL group has been destroyed.")
-        return buf
+        if self._use_communication_streams:
+            import cupy as cp
+
+            self._recv_stream.synchronize()
```

Comment:
Why do we need this?

Reply:
Same as https://github.com/ray-project/ray/pull/47586#discussion_r1792665595

Reply:
Added a comment.